src/main/fc/config.c

```diff
@@ -466,6 +468,21 @@ static void validateAndFixConfig(void)
 #if defined(TARGET_VALIDATECONFIG)
     targetValidateConfiguration();
 #endif
+
+#if defined(USE_OSD)
+    for (int i = 0; i < OSD_TIMER_COUNT; i++) {
+         const uint16_t timer = osdConfig()->timers[i];
+         int src = OSD_TIMER_SRC(timer);
+         int prc = OSD_TIMER_PRECISION(timer);
+         if (src >= OSD_TIMER_SRC_COUNT) {
+             src = 0;
+         }
+         if (prc >= OSD_TIMER_PREC_COUNT) {
+             prc = 0;
+         }
+         osdConfigMutable()->timers[i] = OSD_TIMER(src, prc, OSD_TIMER_ALARM(timer));
+     }
+#endif
```

Comment:
This should be before the block with `targetValidateConfiguration()`.

Also our preferred validation logic would  be to reset an invalid config back to defaults for that element. The defaults for each timer are:
```
    osdConfig->timers[OSD_TIMER_1] = OSD_TIMER(OSD_TIMER_SRC_ON, OSD_TIMER_PREC_SECOND, 10);
    osdConfig->timers[OSD_TIMER_2] = OSD_TIMER(OSD_TIMER_SRC_TOTAL_ARMED, OSD_TIMER_PREC_SECOND, 10);
```